File: py/routes/tags.py
from ..db import model_repo
from ._helpers import json_route, ok


async def _search_tags(request):
    q = request.rel_url.query.get("q", "").strip()
    if not q:
        return ok([])
    tags = await model_repo.search_tags(q)
    return ok(tags)


def add_tag_routes(routes):
    routes.get("/tiny-model-manager/api/tags")(json_route(_search_tags))

File: frontend/src/app/services/tags.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';

const API = '/tiny-model-manager/api';

@Injectable({ providedIn: 'root' })
export class TagService {
  private readonly http = inject(HttpClient);

  searchTags(q: string): Observable<string[]> {
    return this.http
      .get<{ success: boolean; data: string[] }>(`${API}/tags`, { params: { q } })
      .pipe(map((r) => r.data));
  }
}
